### ISSUE TITLE
Design and implement Application Registry

### ISSUE THREAD

**Issue by giolekva** [giolekva/pcloud#14]
Not to confuse with the Application Manager:
* Application Registry (AR) will be a service which indexes and makes application charts registered on different Helm Chart Repositories available installable on local PCloud cluster
* While Application Manager (AM) is responsible for actually installing and managing applications

Users should be able to:
1) Add new Helm Repository location into AR
2) AR should monitor registered repositories for new applications, index them locally expose this information via REST API
3) AM should use AR API to allow users to search and install applications

Helm Repository is described here in detail: https://helm.sh/docs/topics/chart_repository/
In short it is a web server where one can find
1) index.yaml file with the list of all charts and their metadata
2) Actual chart archive files
